src/cmds/qdel.c

```diff
@@ -58,106 +58,212 @@
+			char rmt_server[PBS_MAXDEST + 1];
+
+			/* Check if job was moved to a remote cluster */
+			if (locate_job(p_delstatus->name, clusterid, rmt_server))
+				add_jid_to_list_by_name(p_delstatus->name, rmt_server, rmtlist);
+
+		} else if (p_delstatus->code != PBSE_HISTJOBDELETED) {
+			errtxt = pbse_to_txt(p_delstatus->code);
+			if ((errtxt != NULL) && (p_delstatus->code != PBSE_HISTJOBDELETED)) {
+				fprintf(stderr, "%s: %s %s\n", "qdel", errtxt, p_delstatus->name);
+				any_failed = p_delstatus->code;
+			}
+		}
+	}
+
+	return any_failed;
+}
+
+static int
+get_mail_suppress_count(int connect, int dfltmail, int dfltmailflg)
 {
-	
-	if ((svr == NULL) || (jobid == NULL))
-		return;
-	
-	if (svr->max_sz == 0) {
-		svr->jobids = malloc((NO_OF_JOBIDS + 1) * sizeof(char *));
-		if (svr->jobids == NULL) 
-			goto error;
-		svr->max_sz = NO_OF_JOBIDS;
-	} else if (svr->total_jobs == svr->max_sz) {
-		svr->max_sz *= 2;
-		svr->jobids = realloc(svr->jobids, (svr->max_sz + 1) * sizeof(char *));
-		if (svr->jobids == NULL) 
-			goto error;
+	struct batch_status *ss = NULL;
+	struct batch_status *iter_ss = NULL;
+	char *errmsg;
+	char *keystr, *valuestr;
+
+	if (dfltmailflg == FALSE) {
+		ss = pbs_statserver(connect, NULL, NULL);
+		if (ss == NULL && pbs_errno != PBSE_NONE) {
+			if ((errmsg = pbs_geterrmsg(connect)) != NULL)
+				fprintf(stderr, "qdel: %s\n", errmsg);
+			else
+				fprintf(stderr, "qdel: Error %d\n", pbs_errno);
+			exit(pbs_errno);
+		}
+
+		for (iter_ss = ss; iter_ss != NULL; iter_ss = iter_ss->next) {
+			struct attrl *attr;
+
+			for (attr = ss->attribs; attr != NULL; attr = attr->next) {
+				if (strcmp(attr->name, ATTR_dfltqdelargs) == 0) {
+					if (attr->value != NULL && dfltmailflg != TRUE) {
+						if (parse_equal_string(attr->value, &keystr, &valuestr)) {
+							if (strcmp(keystr, "-Wsuppress_email") == 0) {
+								dfltmail = atol(valuestr);
+								dfltmailflg = TRUE;
+							} else {
+								fprintf(stderr,
+									"qdel: unsupported %s \'%s\'\n",
+									attr->name, attr->value);
+							}
+						}
+					}
+				}
+			}
+		}
 	}
-	
-	svr->jobids[svr->total_jobs++] = jobid;
-	svr->jobids[svr->total_jobs] = NULL;
-	return;
-
-error:
-	fprintf(stderr, "qdel: unable to allocate memory.\n");
-	exit(2);	
+
+	if (ss != NULL)
+		pbs_statfree(ss);
+
+	return dfltmail;
 }
 
 /**
- * @brief
- *	Identify the respective svr_jobid_list struct 
- *  and calls the append_jobid function to append the jobid.
+ * @brief	Helper function to handle job deletion for a given cluster
  *
- * @param[in] job_id - Job id 
- * @param[in] server - server name
- * @param[in,out] svr_jobid_list_hd - head of the svr_jobib_list list
- * 
- * @return int
- * @retval	0	- success
- * @retval	1	- failure
+ * @param[in]	clusterid - id of the cluster, currently the PBS_SERVER value
+ * @param[in]	jobids - the list of jobs to delete on this server
+ * @param[in]	numofjobs - count of jobs to delete
  *
+ * @return int
+ * @retval 0 for Success
```

Comment:
Since you are returning any_failed value, it could potentially be an error code, no?